obp/ope/estimators_tuning.py

```diff
@@ -0,0 +1,956 @@
+# Copyright (c) Yuta Saito, Yusuke Narita, and ZOZO Technologies, Inc. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+"""Off-Policy Estimators with built-in hyperparameter tuning."""
+from dataclasses import dataclass
+from typing import Dict, Optional, List
+
+import numpy as np
+from sklearn.utils import check_scalar
+
+from .estimators import (
+    InverseProbabilityWeighting,
+    DoublyRobust,
+    SwitchDoublyRobust,
+    DoublyRobustWithShrinkage,
+)
+from ..utils import (
+    estimate_confidence_interval_by_bootstrap,
+    check_ope_inputs,
+)
+
+
+@dataclass
+class InverseProbabilityWeightingTuning(InverseProbabilityWeighting):
```

Comment:
(Question)
Would it make sense to have a base class for tuning that takes in an OPE Estimator class and hyperparameters as part of the inputs? 
The implementations for estimators in this file seem almost identical except for the OPE estimator class that is being used to estimate the MSE upper bound, estimate round rewards, etc.